gammapy/cube/psf_map.py

```diff
@@ -0,0 +1,207 @@
+
+        self._psf_map = psf_map
+
+    @property
+    def psf_map(self):
+        """the PSFMap itself (~gammapy.maps.Map)"""
+        return self._psf_map
+
+    @classmethod
+    def read(cls, filename, **kwargs):
+        """ Read a psf_map from file and create a PSFMap object"""
+
+        psfmap = Map.read(filename, **kwargs)
+        return cls(psfmap)
+
+    def write(self, *args, **kwargs):
+        """Write the Map object containing the PSF Library map."""
+        self.psf_map.write(*args, **kwargs)
+
+    def get_energy_dependent_table_psf(self, position):
+        """ Returns EnergyDependentTable PSF at a given position
+
+        Parameters
+        ----------
+        position : `~astropy.coordinates.SkyCoord`
+            the target position. Should be a single coordinates
+
+        Returns
+        -------
+        psf_table : `~gammapy.irf.EnergyDependentTablePSF`
+            the table PSF
+        """
+        if position.size != 1:
+            raise ValueError("EnergyDependentTablePSF can be extracted at one single position only.")
+
+        # axes ordering fixed. Could be changed.
+        pix_ener = np.arange(self.psf_map.geom.axes[1].nbin)
+        pix_rad = np.arange(self.psf_map.geom.axes[0].nbin)
+
+        # Convert position to pixels
+        pix_lon, pix_lat = self.psf_map.geom.to_image().coord_to_pix(position)
+
+        # Build the pixels tuple
+        pix = np.meshgrid(pix_lon, pix_lat,pix_rad,pix_ener)
+
+        # Interpolate in the PSF map. Squeeze to remove dimensions of length 1
+        psf_values = np.squeeze(self.psf_map.interp_by_pix(pix)*u.Unit(self.psf_map.unit))
+
+        energies = self.psf_map.geom.axes[1].center * self.psf_map.geom.axes[1].unit
+        rad = self.psf_map.geom.axes[0].center * self.psf_map.geom.axes[0].unit
+
+        # Beware. Need to revert rad and energies to follow the TablePSF scheme.
+        return EnergyDependentTablePSF(energy=energies,rad=rad,psf_value=psf_values.T)
+
+    def get_psf_kernel(self, position, geom, max_radius=None, factor=4):
+        """Returns a PSF kernel at the given position.
+
+        The PSF is returned in the form a WcsNDMap defined by the input MapGeom.
+
+        Parameters
+        ----------
+        position : `~astropy.coordinates.SkyCoord`
+            the target position. Should be a single coordinate
+        geom : `~gammapy.maps.MapGeom`
+            the target geometry to use
+        max_radius: `~astropy.coordinates.Angle`
+            maximum angular size of the kernel map
+        factor : int
+            oversampling factor to compute the PSF
+
+        Returns
+        -------
+        kernel : `~gammapy.cube.PSFKernel`
+            the resulting kernel
+        """
+        table_psf = self.get_energy_dependent_table_psf(position)
+        return PSFKernel.from_table_psf(table_psf, geom, max_radius, factor)
+
+    def containment_radius_map(self, fraction = 0.68):
```

Comment:
How much work is it to implement the containment radius map?
Would be very nice to have, no?
Alternatively, please open an issue for this and label it beginner friendly and we'll try to find soemone to do it next week.